### PR TITLE
Add support for Smartmi Air Purifier (zhimi.airpurifier.za1)

### LINKED ISSUE #1025: Add new device support - Smartmi Air Purifier
### Request Smartmi zhimi.airpurifer.za1 - Air Purifier support 
[](https://www.smartmiglobal.com/collections/air-purifier/products/smartmi-air-purifier
)

**Xiaomi App Detalis details:**

```
ID: xxx
IP: 192.168.0.xx
MODEL: zhimi.airpurifer.za1
```

### PR DESCRIPTION
**Pull request by julian-klode**
Add support for zhimi.airpurifier.za1

This also adds a tvoc attribute, and the gesture control setting in a 2nd commit.

Fixes #1025

### CODE REVIEW

**Comment by julian-klode** [comment]
I don't know what unit the TVOC value is, it seems to go from 0..500 in 1 steps according to the spec.

There's a bunch more attributes like max. filter time, when the filter door was last opened that I did not bother adding as they do not seem all that useful.

**Comment by julian-klode** [comment]
flake8 failure seems unrelated flake8 issue, should be fixed in master first and I guess then rebase the PR.

I still see:

WARNING:miio.device:Found an unsupported model 'zhimi.airpurifier.za1' for class 'AirPurifierMiot'. If this is working for you, please open an issue at https://github.com/rytilahti/python-miio/

But I'm not sure where else to put that string to get it to be recognized.

File: miio/integrations/airpurifier/zhimi/airpurifier_miot.py [hunk 1]
```diff
@@ -515,6 +567,23 @@ def set_buzzer(self, buzzer: bool):
...
+            lambda gestures: "Turning on gestures"
+            if gestures
+            else "Turning off gestures"
+        ),
+    )
+    def set_gestures(self, gestures: bool):
+        """Set gestures on/off."""
+        if "gestures" not in self._get_mapping():
+            raise AirPurifierMiotException(
+                "Unsupported buzzer for model '%s'" % self.model
```
Comment (rytilahti) [inline miio/integrations/airpurifier/zhimi/airpurifier_miot.py:582]:
```suggestion
                "Gestures not support for model '%s'" % self.model
```

**Comment by julian-klode** [comment]
I'm also confused about the warning. It happens with a clean-ish docker container running a tiny python script, so hmm (e9d9270cfe14445e7a16eae4e0a3fa0fb675995a being the previous head of the branch).

```
jak@jak-t480s:~/Projects/Stable/python-miio:zhimi.airpurifier.za1$ cat tvoc 
#!/usr/bin/python3
import miio
import time
import paho
import sys

ap=miio.AirPurifierMiot("<ip>", "<key>")


import paho.mqtt.client

import miio

client=paho.mqtt.client.Client()
client.connect("192.168.0.233")


while True:
    try:
        print("Request", file=sys.stderr)
        tvoc = ap.status().tvoc
        print(time.strftime("%H:%M:%S", time.localtime()), tvoc, file=sys.stderr)
        client.publish("sensors/tvoc", tvoc)
    except miio.exceptions.DeviceException as exc:
        print(time.strftime("%H:%M:%S", time.localtime()), exc, file=sys.stderr)
        pass

    time.sleep(5)
jak@jak-t480s:~/Projects/Stable/python-miio:zhimi.airpurifier.za1$ cat Dockerfile 
FROM ubuntu:jammy
RUN apt update && apt install -y python3-pip
RUN pip3 install --upgrade pip poetry
WORKDIR /miio
RUN /usr/lib/apt/apt-helper download-file https://github.com/julian-klode/python-miio/tarball/e9d9270cfe14445e7a16eae4e0a3fa0fb675995a /tmp/miio.tar.gz && tar xvzf /tmp/miio.tar.gz -C /miio --strip-components=1 && rm /tmp/miio.tar.gz
RUN poetry install
RUN poetry add paho-mqtt
ADD tvoc /usr/bin
ENTRYPOINT poetry run python /usr/bin/tvoc
```

Running this yields
```
Jun 21 07:12:58 iot-gateway docker[226357]: Request
Jun 21 07:12:59 iot-gateway docker[226357]: Found an unsupported model 'zhimi.airpurifier.za1' for class 'AirPurifierMiot'. If this is working for you, please open an issue at https://github.com/rytilahti/python-miio/
```

Not sure what else would be missing to drop this, but it works fine otherwise. And for really unsupported models one would additionally get:
```
WARNING:miio.miot_device:Unable to find mapping for zhimi.airpurifier.za1, falling back to xiaomi.aircondition.mc1
```

**Review by rytilahti — APPROVED** [review]
LGTM, let's get this into the next release, thanks @julian-klode! :+1:

**Pull request merged**